commit f1fe12c8bf33241e04c57a0fc5b25b16ba77ba2d
Author: Linus Walleij <linus.walleij@linaro.org>
Date:   Thu Feb 15 16:12:29 2018 +0100

    ARM: dts: Modernize the Vexpress PL111 integration
    
    The Versatile Express was submitted with the actual display
    bridges unconnected (but defined in the device tree) and
    mock "panels" encoded in the device tree node of the PL111
    controller.
    
    This doesn't even remotely describe the actual Versatile
    Express hardware. Exploit the SiI9022 bridge by connecting
    the PL111 pads to it, making it use EDID or fallback values
    to drive the monitor.
    
    The  also has to use the reserved memory through the
    CMA pool rather than by open coding a memory region and
    remapping it explicitly in the driver. To achieve this,
    a reserved-memory node must exist in the root of the
    device tree, so we need to pull that out of the
    motherboard .dtsi include files, and push it into each
    top-level device tree instead.
    
    We do the same manouver for all the Versatile Express
    boards, taking into account the different location of the
    video RAM depending on which chip select is used on
    each platform.
    
    This plays nicely with the new PL111 DRM driver and
    follows the standard ways of assigning bridges and
    memory pools for graphics.
    
    Cc: Sudeep Holla <sudeep.holla@arm.com>
    Cc: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
    Cc: Liviu Dudau <liviu.dudau@arm.com>
    Cc: Mali DP Maintainers <malidp@foss.arm.com>
    Cc: Robin Murphy <robin.murphy@arm.com>
    Tested-by: Liviu Dudau <liviu.dudau@arm.com>
    Signed-off-by: Linus Walleij <linus.walleij@linaro.org>

diff --git a/arch/arm64/boot/dts/arm/rtsm_ve-motherboard.dtsi b/arch/arm64/boot/dts/arm/rtsm_ve-motherboard.dtsi
index d2dbc3f39263..b25f3cbd3da8 100644
--- a/arch/arm64/boot/dts/arm/rtsm_ve-motherboard.dtsi
+++ b/arch/arm64/boot/dts/arm/rtsm_ve-motherboard.dtsi
@@ -24,11 +24,6 @@ flash@0,00000000 {
 				bank-width = <4>;
 			};
 
-			v2m_video_ram: vram@2,00000000 {
-				compatible = "arm,vexpress-vram";
-				reg = <2 0x00000000 0x00800000>;
-			};
-
 			ethernet@2,02000000 {
 				compatible = "smsc,lan91c111";
 				reg = <2 0x02000000 0x10000>;
@@ -187,38 +182,16 @@ clcd@1f0000 {
 					interrupts = <14>;
 					clocks = <&v2m_oscclk1>, <&v2m_clk24mhz>;
 					clock-names = "clcdclk", "apb_pclk";
-					arm,pl11x,framebuffer = <0x18000000 0x00180000>;
-					memory-region = <&v2m_video_ram>;
-					max-memory-bandwidth = <130000000>; /* 16bpp @ 63.5MHz */
+					/* 800x600 16bpp @36MHz works fine */
+					max-memory-bandwidth = <54000000>;
+					memory-region = <&vram>;
 
 					port {
-						v2m_clcd_pads: endpoint {
-							remote-endpoint = <&v2m_clcd_panel>;
+						clcd_pads: endpoint {
+							remote-endpoint = <&panel_in>;
 							arm,pl11x,tft-r0g0b0-pads = <0 8 16>;
 						};
 					};
-
-					panel {
-						compatible = "panel-dpi";
-
-						port {
-							v2m_clcd_panel: endpoint {
-								remote-endpoint = <&v2m_clcd_pads>;
-							};
-						};
-
-						panel-timing {
-							clock-frequency = <63500127>;
-							hactive = <1024>;
-							hback-porch = <152>;
-							hfront-porch = <48>;
-							hsync-len = <104>;
-							vactive = <768>;
-							vback-porch = <23>;
-							vfront-porch = <3>;
-							vsync-len = <4>;
-						};
-					};
 				};
 
 				virtio-block@130000 {